kernel/section.mli

```diff
@@ -0,0 +1,71 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *   INRIA, CNRS and contributors - Copyright 1999-2019       *)
+(* <O___,, *       (see CREDITS file for the list of authors)           *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+open Names
+open Univ
+
+(** Kernel implementation of sections. *)
+
+type t
+(** Type of sections *)
```

Comment:
I'm not sure I like the interface a lot, in the sense that there are too many cases in the code that are different whether sections are opened or not. Ideally tho the default style of the kernel should be to assume there is always a section open as this is what most large developments tend to work with, but that'd require a deep design of the typechecking. A few comments then:
- You could use a module `Stack` to properly scope some of the functions here.
- `Section.t` should not allow for non-open sections, that is to say, code that can work with and without section data should take a `Section.t option` in the same way proof code does. That would remove all these `user_err`.
- The way stacking is done seems weird? See below.